monopoly/src/main/java/gameset/functionality/Game.java

```diff
@@ -66,21 +66,31 @@ private void listPlayers() {
      * @param player the player object
      * @param property object which the building will be added to
      */
-    private void purchaseBuildings(Property property, Player player) {
-        if (player.canPurchaseBuilding(property.getColor())) {
-            int cost = property.getBuildingPrice();
-            if (player.getMoney() - cost >= 0) {
-                player.setMoney(player.getMoney() - cost);
-                property.buildBuilding();
-                System.out.println("You purchased your " + property.getNumHouses() + " houses of " + property.displayPropertyName());
-            }
-            else {
-                System.out.println("Unable to purchase building due to insufficient funds");
-            }
+    private void purchaseBuildings(Property property, Player player) throws IOException {
+        if (!player.canPurchaseBuilding(property, board)) {
+            System.out.println("Unable to purchase building due to not having all properties of the color set for: " + property.getColor());
+            return;
         }
-        else {
-            System.out.println("Unable to purchase building due to you not having all colorset for: " + property.getColor());
+
+        int cost = property.getBuildingPrice();
+
+        if (property.isMortgaged()) {
+            System.out.println("Unable to purchase buildings on a mortgaged property.");
+            return;
+        }
+
+        if (player.getMoney() >= cost) {
+            System.out.println("Unable to purchase building due to insufficient funds.");
+            return;
         }
+
+        if (!property.buildBuilding(board)) {
+            System.out.println("No more buildings allowed to be purchased.");
+            return;
+        }
+
+        player.setMoney(player.getMoney() - cost);
+        System.out.println("You purchased " + property.getNumHouses() + " house(s) on " + property.displayPropertyName());
```

Comment:
The wording might be misleading here. I would think I purchased x amount of houses. This looks to be returning the new total amount of houses on the property, right?

```suggestion
        System.out.println("You now have " + property.getNumHouses() + " house(s) on " + property.displayPropertyName());
```